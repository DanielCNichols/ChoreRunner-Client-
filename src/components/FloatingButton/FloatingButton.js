import React, { useState } from 'react';
import cx from 'classnames';
import s from './FloatingButton.module.css';
import { BsPlusCircleFill } from 'react-icons/bs';

const FloatingButton = ({ className }) => {
  return (
    <div className={s.buttonMenuContainer}>
      <button className={cx(`${s.FloatingButton}`, className)}>
        <BsPlusCircleFill />
      </button>
    </div>
  );
};

export default FloatingButton;
