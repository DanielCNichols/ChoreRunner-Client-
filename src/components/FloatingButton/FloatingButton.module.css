@import '../../vars.css';

.FloatingButton {
  border: 2px solid white;
  border-radius: 50%;
  outline: none;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 65px;
  height: 65px;
  padding: 0;
  position: fixed;
  bottom: 15%;
  right: 15%;
}

.FloatingButton svg {
  width: 100%;
  height: 100%;
  background-color: white;
  color: var(--color-mainHeader);
  border-radius: inherit;
}

@media screen and (min-width: 800px) {
  .FloatingButton {
    display: none;
  }
}
