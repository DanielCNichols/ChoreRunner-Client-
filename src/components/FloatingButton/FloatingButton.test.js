import React from 'react';
import ReactDOM from 'react-dom';
import FloatingButton from './FloatingButton';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Button component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <MemoryRouter>
        <FloatingButton />
      </MemoryRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <FloatingButton />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
