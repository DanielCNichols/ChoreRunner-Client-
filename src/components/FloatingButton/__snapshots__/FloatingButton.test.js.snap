// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component testing renders UI as expected 1`] = `
<div
  className="buttonMenuContainer"
>
  <button
    className="FloatingButton"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M16 8A8 8 0 110 8a8 8 0 0116 0zM8.5 4a.5.5 0 00-1 0v3.5H4a.5.5 0 000 1h3.5V12a.5.5 0 001 0V8.5H12a.5.5 0 000-1H8.5V4z"
        fillRule="evenodd"
      />
    </svg>
  </button>
</div>
`;
